File: internal/content/testdata/bench/go/nc-group-anagrams.go
import (
	"sort"
)

func group_anagrams(strs []string) [][]string {
	groups := make(map[string][]string)
	for _, s := range strs {
		runes := []byte(s)
		sort.Slice(runes, func(i, j int) bool { return runes[i] < runes[j] })
		key := string(runes)
		groups[key] = append(groups[key], s)
	}
	res := make([][]string, 0, len(groups))
	for _, g := range groups {
		sorted := make([]string, len(g))
		copy(sorted, g)
		sort.Strings(sorted)
		res = append(res, sorted)
	}
	sort.Slice(res, func(i, j int) bool {
		for k := 0; k < len(res[i]) && k < len(res[j]); k++ {
			if res[i][k] != res[j][k] {
				return res[i][k] < res[j][k]
			}
		}
		return len(res[i]) < len(res[j])
	})
	return res
}

File: internal/content/testdata/bench/rust/nc-group-anagrams.rs
fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
    use std::collections::HashMap;
    let mut groups: HashMap<String, Vec<String>> = HashMap::new();
    for s in &strs {
        let mut chars: Vec<char> = s.chars().collect();
        chars.sort();
        let key: String = chars.into_iter().collect();
        groups.entry(key).or_default().push(s.clone());
    }
    let mut res: Vec<Vec<String>> = groups
        .into_values()
        .map(|mut g| {
            g.sort();
            g
        })
        .collect();
    res.sort();
    res
}
